corehq/apps/accounting/models.py

```diff
@@ -1230,6 +1232,13 @@ def send_ending_reminder_email(self):
                     'email': email,
                 })
 
+    def set_billing_account_entry_point(self):
+        no_current_entry_point = self.account.entry_point == EntryPoint.NOT_SET
+        self_serve = self.service_type == SubscriptionType.SELF_SERVICE
+        if (no_current_entry_point and self_serve and not self.is_trial):
+            self.account.entry_point = EntryPoint.SELF_STARTED
+            self.account.save()
```

Comment:
I was surprised that this method saves the account - both because it means sometimes saving twice when an account is created/renewed, and because of the potential for someone to call this and not realize it would save any modifications.